lib/acpi/acpi.c

```diff
@@ -289,14 +289,13 @@ int acpi_current_resource_get(char *dev_name, ACPI_RESOURCE **res)
 
 	status = check_init_status();
 	if (status) {
-		return status;
```

Comment:
the check_init_status() itself can return mixed error code

Reply:
> the check_init_status() itself can return mixed error code

Good catch. Will fix.

Reply:
> > the check_init_status() itself can return mixed error code
> 
> Good catch. Will fix.

Fixed